prov/efa/test/efa_unit_test_runt.c

```diff
@@ -0,0 +1,277 @@
+#include "efa_unit_tests.h"
+
+/**
+ * @brief Test the runt size returned by efa_rdm_peer_get_runt_size
+ *
+ * @param resource efa_resource
+ * @param iface hmem iface
+ * @param peer_num_runt_bytes_in_flight how many runt bytes are in flight on the peer side
+ * @param total_runt_size the total runt size
+ * @param total_len the total length of the message to be sent
+ * @param expected_runt_size the expected runt size
+ */
+static
+void test_efa_rdm_peer_get_runt_size_impl(
+		struct efa_resource *resource,
+		enum fi_hmem_iface iface, size_t peer_num_runt_bytes_in_flight,
+		size_t total_runt_size, size_t total_len, size_t expected_runt_size)
+{
+	struct efa_rdm_ep *efa_rdm_ep;
+	struct efa_ep_addr raw_addr;
+	size_t raw_addr_len = sizeof(raw_addr);
+	struct efa_rdm_peer *peer;
+	fi_addr_t addr;
+	struct efa_mr mock_mr;
+	struct efa_rdm_ope mock_txe;
+	size_t runt_size;
+	struct efa_domain *efa_domain;
+	int ret;
+
+	efa_rdm_ep = container_of(resource->ep, struct efa_rdm_ep, base_ep.util_ep.ep_fid);
+	efa_domain = efa_rdm_ep_domain(efa_rdm_ep);
+	efa_domain->hmem_info[iface].runt_size = total_runt_size;
+
+	/* insert a fake peer */
+	ret = fi_getname(&resource->ep->fid, &raw_addr, &raw_addr_len);
+	assert_int_equal(ret, 0);
+	raw_addr.qpn = 1;
+	raw_addr.qkey = 0x1234;
+	ret = fi_av_insert(resource->av, &raw_addr, 1, &addr, 0 /* flags */, NULL /* context */);
+	assert_int_equal(ret, 1);
+	peer = efa_rdm_ep_get_peer(efa_rdm_ep, addr);
+	peer->num_runt_bytes_in_flight = peer_num_runt_bytes_in_flight;
+
+	mock_mr.peer.iface = iface;
+
+	memset(&mock_txe, 0, sizeof(mock_txe));
+	mock_txe.total_len = total_len;
+	mock_txe.addr = addr;
+	mock_txe.iov_count = 1;
+	mock_txe.iov[0].iov_base = NULL;
+	mock_txe.iov[0].iov_len = 9000;
```

Comment:
`mock_txe.iov[0].iov_len = total_len` or simply skip initializing `mock_txe.iov`

Same in `test_efa_rdm_peer_select_readbase_rtm_impl`

Reply:
Done